sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -337,9 +346,23 @@ def _delete(self, **kwargs):
         :returns: A dictionary containing information about the deleted path.
         :rtype: Dict[str, Any]
         """
-        options = self._delete_path_options(**kwargs)
+        # Perform paginated delete only if using OAuth and api version is 2023-08-03 or later
+        # The pagination is only for ACL checks, the final requests remains the atomic delete operation
+        paginated = None
+        if (compare_api_versions(self.api_version, '2023-08-03') >= 0 and
+            hasattr(self.credential, 'get_token')):
```

Comment:
What would be the behaviour if we didn't perform this check?
I'm just wondering whether this falls into the bucket of client-side validation we'd rather avoid... Or if it's more aligned with optimization.... seems more like the latter given that we still fallback to unpaginated...

Reply:
- For the API version check, requests will fail if `paginated` is provided on older versions.
- For the OAuth, I don't believe the request will fail if we pass this with other auth types but pagination only every occurs and therefore makes sense with OAuth today. So, it's sort of an optimization we don't pass it and also, I think it's good in case the service decides to start doing things differently with it for other auth types or something.
- In the latest revision I will also add another check here for the `recursive` keyword to be true. This is to ensure we only call this on directory delete and not file delete (since this is the shared `PathClient`) as the request will fail on file delete. We hard-code `recursive=True` for directory delete and don't provide it for file delete.